util.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+)
+
+// SchemaDebug provides a logging interface
+// which is off by defauly but can be activated
+// for debuging purposes
+func SchemaDebug(message string) {
+	debug := false
+	debugEnvVar := os.Getenv("JSON_SCHEMA_DEBUG")
+	if debugEnvVar == "1" {
+		debug = true
+	}
+	if debug {
+		fmt.Printf("%s\n", message)
+	}
+}
+
+// SafeResolveURL resolves a string url against the current context url
+func SafeResolveURL(ctxURL, resURL string) (string, error) {
+	cu, err := url.Parse(ctxURL)
+	if err != nil {
+		return "", err
+	}
+	u, err := url.Parse(resURL)
+	if err != nil {
+		return "", err
+	}
+	resolvedURL := cu.ResolveReference(u)
+	if resolvedURL.Scheme == "file" && cu.Scheme != "file" {
+		return "", fmt.Errorf("cannot access file resources from network context")
+	}
+	resolvedURLString := resolvedURL.String()
+	return resolvedURLString, nil
+}
+
+// IsLocalSchemaID validates if a given id is a local id
+func IsLocalSchemaID(id string) bool {
+	splitID := strings.Split(id, "#")
+	if len(splitID) > 1 && len(splitID[0]) > 0 && splitID[0][0] != '#' {
+		return false
+	}
+	return id != "#" && !strings.HasPrefix(id, "#/") && strings.Contains(id, "#")
+}
+
+// FetchSchema downloads and loads a schema from a remote location
+func FetchSchema(ctx *context.Context, uri string, schema *Schema) error {
```

Comment:
don't use pointers to context.Context values: 
```suggestion
func FetchSchema(ctx context.Context, uri string, schema *Schema) error {
```